### ISSUE TITLE
[Bug]: Integration not updating Armed/Disarmed Status

### ISSUE THREAD

**Issue by Hansontech190** [bvis/aegis-hass#46]
### Describe the bug

I have the alarm setup to arm night mode on a schedule. The system also disarms on a schedule. 

The integration is showing Night Armed despite it being disarmed this morning. 

### Steps to reproduce

Arm system. Disarm system and integration shows armed. 

### Expected behavior

Active reflection of alarm state. 

### Home Assistant version

2026.4.3

### Integration version

1.2.0 Beta 2

### Installation method

HACS

### App label (co-branded app)

Ajax

### FCM push notifications configured?

Yes

### Relevant logs

```text

```

**Comment by martin-gawlita** [comment]
I've just added the integration and also found out, that the armed state is just updated if it's updated by the integration. If it's armed by another device or account. I don't see any state changes in HA

**Comment by bvis** [comment]
@martin-gawlita — quick question first: have you configured **FCM push notifications** in the integration's options? Without FCM, the integration only learns about state changes via the 5-minute fallback poll, so anything armed/disarmed by another device, by the Ajax cloud scheduler, or by another user would take up to 5 min to surface in HA. State changes triggered *from* HA refresh immediately because the action triggers a refresh, which would explain the asymmetry you're seeing. If FCM is not set up, that's the most likely cause for your case.

@Hansontech190 — your issue is different (FCM configured, arm/disarm from a schedule). To narrow it down we need to see what the integration is doing during one full schedule-driven arm → disarm cycle. Could you enable debug logging and capture it?

**Option A — no restart needed**: HA → *Developer Tools* → *Actions* → pick `logger.set_level` → switch to YAML mode and paste:

```yaml
custom_components.aegis_ajax: debug
custom_components.aegis_ajax.coordinator: debug
custom_components.aegis_ajax.notification: debug
custom_components.aegis_ajax.api.spaces: debug
```

**Option B — persistent**: add to `configuration.yaml` and restart HA:

```yaml
logger:
  default: info
  logs:
    custom_components.aegis_ajax: debug
    custom_components.aegis_ajax.coordinator: debug
    custom_components.aegis_ajax.notification: debug
    custom_components.aegis_ajax.api.spaces: debug
```

Then:
1. Wait for the scheduler to fire (or manually trigger the arm/disarm via the Ajax app).
2. Note the time of the schedule action and what HA showed at that moment.
3. Download the log: *Settings → System → Logs → Download full log*.
4. Attach it here. The relevant lines start with `custom_components.aegis_ajax`. Feel free to redact device/space ids.

Also, if you can: please confirm you're still on `1.2.0 beta 2` or if you've moved to a newer build (`1.2.0` stable, `1.2.1-beta.*`). The push-notification path got refactored across those, so the version matters.

**Comment by martin-gawlita** [comment]
> [@martin-gawlita](https://github.com/martin-gawlita) — quick question first: have you configured **FCM push notifications** in the integration's options? Without FCM, the integration only learns about state changes via the 5-minute fallback poll, so anything armed/disarmed by another device, by the Ajax cloud scheduler, or by another user would take up to 5 min to surface in HA. State changes triggered _from_ HA refresh immediately because the action triggers a refresh, which would explain the asymmetry you're seeing. If FCM is not set up, that's the most likely cause for your case.
> 

Thank you for the quick answer and description. Haven't seen this setting ans will test it today.

**Comment by Hansontech190** [comment]
Hi @bvis 

I've enabled logging and then night armed from Ajax app.  After waiting around 10mins the HA interface showed disarmed and never changed to armed.  Tried it a second time with same result.  Hopefully logs attached but not showing much I can see.  Can arm/disarm from HA without issue.  I'm still on 1.2.0 beta 2.

[HA Log Ajax.rtf](https://github.com/user-attachments/files/27118124/HA.Log.Ajax.rtf)

**Comment by K0rtex79** [comment]
Hi same here, can arm / disarm from integration but no change in HA even after 15 min wait.
FCM not configured

**Comment by bvis** [comment]
Thanks all for the extra info. Looking at the log Hansontech190 attached, I can see one important thing right away:

```
[custom_components.aegis_ajax.notification] FCM credentials not configured, push notifications disabled
```

So even though the issue form says "FCM configured: Yes", the actual integration started with FCM disabled. @martin-gawlita and @K0rtex79 also explicitly mention not having FCM. That's the common factor across all three reports.

### What's happening

Without FCM, the integration only learns about state changes initiated outside HA via the **5-minute fallback poll** (`coordinator.py` calls `SpaceService/findUserSpacesWithPagination` once per cycle and refreshes `coordinator.spaces[...].security_state`). Changes initiated **from HA** refresh instantly because the action itself triggers a refresh and we hold an optimistic state for 10s to absorb stale server replies. That's the asymmetry you're seeing — it's expected behaviour for non-FCM setups, with up to 5 min of lag.

### What's not expected

@Hansontech190 reports waiting 10 minutes (≈ 2 poll cycles) and HA never reflecting the change. The captured log is just ~1 second of startup activity so there's no actual poll cycle in it, but a likely cause is that you're on **`1.2.0 beta 2`**, which is older than the fix in `1.2.1-beta.4` for stale-token recovery: if Ajax invalidated the integration's session (which it does when too many active sessions accumulate), the old code would raise `UpdateFailed` on every cycle and never refresh state until you restarted HA — symptom indistinguishable from "the integration just stops updating".

### What to do

1. **Upgrade to `1.2.1-beta.8`** (latest beta). It includes the UNAUTHENTICATED-recovery fix plus several stream-handling improvements.
2. Then either:
   - **Configure FCM** — gives you real-time updates for events. The FCM API key for the standard Ajax app is usually in `res/values/strings.xml`, see the README's "Push Notifications" section.
   - **Lower the poll interval** in *Settings → Integrations → Aegis for Ajax → Configure → Poll interval* (minimum 60s). Tradeoff: more API calls, but worst-case lag drops from 5 min to whatever you set.
3. If after upgrading you still see "never updates", please capture a debug log that **covers a full change cycle** (your previous one was only the first second after restart): start logging, wait until the schedule fires *and* at least 10 minutes after, then download. The relevant lines start with `custom_components.aegis_ajax.coordinator` — look for `Finished fetching aegis_ajax data` (success) or any `UpdateFailed` (stuck).

Separately, I've opened #68 to track an enhancement that will close the gap for FCM users — apply the new `security_state` from the arm/disarm push directly, without waiting for the next poll.

**Comment by bogar** [comment]
Polling every 5m works for me in 1.2.1-beta.6 👍

**Comment by Hansontech190** [comment]
Good Morning.

Unfortunately it's not working for me on 1.2.1 beta 8.  I thought it had worked as it seemed to show night mode and disarmed this morning but the times were incorrect.

<img width="567" height="337" alt="Image" src="https://github.com/user-attachments/assets/5937d67d-b25e-4a51-b7b9-51b833663955" />

I did a manual night set at 0909 this morning and then waited and watched the live logs.  No change to the status on HA even after 5 mins - I haven't reduced polling time yet, still set at 300s.  I then disramed at 0918 and again watched the live logs.  All attached including from last night.  I hope the attached is a better record of the logs.

If it's working for others must be something in my setup.

[HA Log Ajax.rtf](https://github.com/user-attachments/files/27157091/HA.Log.Ajax.rtf)

**Comment by bvis** [comment]
Thanks for the log @Hansontech190 — it pointed us at something different from what we'd assumed. Lowering the poll interval or enabling FCM wouldn't have helped in your case; the integration's update cycle is getting stuck instead of running on schedule. Tracking it dedicatedly in #74 so it gets the attention it needs.

In the meantime, **reloading or restarting the integration** (Settings → Devices & Services → Aegis for Ajax → ⋮ → Reload) once a day works around it until the underlying fix lands. Leaving this issue open as the user-facing symptom; once #74 has a fix we'll wire them together.

**Comment by Hansontech190** [comment]
Thanks @bvis. Appreciate the prompt reply. 

Sorry to ask here but if I wanted to use FCM how do I obtain the details when running Ajax app on iOS?  I know it's relatively easy on Android but not sure how to obtain necessary details on iOS.

**Comment by bvis** [comment]
Closing this thread now that we've localised what's actually behind the symptom — turns out the same "panel doesn't update from outside HA" report has two distinct root causes that were getting mixed up here:

- **Without FCM configured**: the `coordinator` polling cycle is stalling for hours instead of running on schedule, so `coordinator.spaces[...].security_state` never refreshes regardless of how short you set the poll interval. Tracking that dedicatedly in **#74** with a working hypothesis and a fix direction. If you don't have FCM set up, that's the issue to follow.
- **With FCM configured**: in some co-branded builds the parser doesn't recognise the arm/disarm push payload, so the in-memory shortcut introduced in `1.2.1-beta.10` never fires. That's tracked in **#68**.

In the meantime the practical workaround is **reload the integration once a day** (Settings → Devices & Services → Aegis for Ajax → ⋮ → Reload) — that resets the coordinator state and gets polls flowing again until the underlying fix lands.

Thanks @Hansontech190, @martin-gawlita, @K0rtex79 and @bogar — your logs and confirmations were what made it possible to split this cleanly. Please subscribe to #74 (and #68 if you have FCM wired) for updates, and feel free to reopen here if you see a third distinct pattern that doesn't match either.

**Comment by jjeremia** [comment]
I'm seeing the same issue and would like to add some data points in case this gets reopened — the failure mode I'm experiencing matches the original report exactly.

## Summary

The integration silently stops pushing state updates from the Ajax hub after some hours of normal operation. The config entry remains `state: loaded`, no errors are logged, and the integration appears healthy in the UI — but `alarm_control_panel.<name>` stays frozen at its last received value and stops reflecting arm/disarm operations done at the physical key panel.

## Environment

- **Home Assistant Core:** 2026.4.4 (Home Assistant OS 17.2)
- **Architecture:** amd64 (KVM virtualized)
- **Python:** 3.14.2
- **Supervisor:** 2026.04.1
- **Integration source:** HACS
- **Integration version:** [please fill in — visible in HACS]
- **Hub:** Ajax Hub Two Plus
- **App label:** Ajax
- **FCM push notifications configured:** Yes
- **Authentication:** OAuth (Ajax Security cloud login)
- **Config entry age at first failure:** ~6 weeks

## Timeline from my most recent occurrence

- Last successful state update received by HA: 2026-04-29 09:36 UTC (`armed_night -> disarmed`).
- Multiple physical arm/disarm operations on the panel after this point — none reflected in HA.
- Issue noticed at 2026-04-29 21:32 UTC (~12 hours of silence).
- Disabled and re-enabled the config entry via the UI — entity refreshed to `armed_night` instantly, matching the panel's actual state.
- Normal state-change events resumed.

This matches the original report: the integration goes silent and the only way to recover is manual intervention.

## What's notable about the failure

- The config entry continues to report `state: loaded`, `disabled_by: null`. No error is surfaced in **Settings -> Repairs**.
- Searching `system_log`, `error_log`, and supervisor core logs for `aegis`, `ajax`, or the entity name returns **zero entries** during or after the failure. The integration goes silent without raising anything.
- Polling is enabled (`pref_disable_polling: false`).
- This is not the first time it's happened — I noticed today after the same symptom appeared before.

## Workaround

Disable and re-enable the config entry (Settings -> Devices & Services -> Ajax Security -> three-dot menu -> Disable, then Enable). No HA restart required. Forces a fresh setup including new auth tokens.

## Suspected cause (without source access)

- OAuth access token expiring without proper refresh.
- Long-lived websocket / SSE connection to Ajax cloud being closed by the server (idle timeout, server-side push, network blip) without the client reconnecting.
- An unhandled exception in the update coordinator getting swallowed, leaving the coordinator in a stale state.

## Suggested improvements

1. Heartbeat / liveness check in the update coordinator: if no events have arrived within an expected window, trigger a reconnect.
2. Wrap the connection loop in robust exception handling that logs at WARNING and transitions the config entry to `setup_retry` so HA's standard retry mechanism takes over.
3. Expose a `last_seen` or `connection_status` diagnostic sensor so users can build watchdog automations until a permanent fix lands.
4. If OAuth refresh is the culprit: refresh proactively before expiry, not lazily on the next failed request.

Happy to enable DEBUG logging on `aegis_ajax` and provide a diagnostics download from the config entry, or run any specific reproduction steps that would help. Could this issue be reopened?

**Comment by jjeremia** [comment]
What's the intended way for users to obtain the FCM credentials? Is there documentation, or is this intended only for users of certain co-branded apps?

**Comment by Hansontech190** [comment]
This problem remains. I'm seeing very long delays between the system being changed and the status updating in HA - hours of delay. 

Can this be looked at again please.

**Comment by bvis** [comment]
@Hansontech190 @jjeremia @K0rtex79 — picking this back up.

### Where the two known causes stand

Both fixes the original closure pointed at shipped in **1.2.2 stable** and have been in main ever since:

- **#68** (FCM payload not parsed for some co-branded apps) → `04452df`: parse `SpaceEventQualifier` in the arm/disarm push so FCM users see state in real time.
- **#74** (coordinator polling stalls for hours, no log entries) → `bb5567e`: bound the HTS auth handshake with a hard timeout so the coordinator can't get wedged mid-handshake.

@Hansontech190 — your original report was on `1.2.0-beta.2`, and the #74 symptom you described ("HA never reflects the change, no error in logs") matches that build exactly. **First step on your side**: upgrade to the latest beta (`1.2.4-beta.6`) and see if the symptom persists. If you're already on 1.2.2+, see the diagnostics block below.

### What the 1.2.4 line added that's directly relevant

The current beta (`1.2.4-beta.6`) ships three things specifically aimed at making this class of "integration goes silent" easier to **see** and to self-recover from, addressing several of the suggestions in @jjeremia's writeup:

- **System Health card** (*Settings → System → System Information*) shows for each Aegis account: gRPC reachability, HTS streams alive (`N/M`), FCM clients alive, total pushes received, **age of last push received**, and **age of last successful poll**. That's the `last_seen`/heartbeat diagnostic you asked for.
- **HA Repairs** that surface the silent-stall conditions as actionable cards:
  - `hub_offline_24h` — hub OFFLINE on consecutive snapshots for 24h+, clears on reconnect.
  - `hts_chronic_failure` — HTS reconnect failing for 30 min straight, clears on next successful reconnect.
  - `fcm_credentials_invalid` — Firebase rejected the configured api_key/app_id/project_id/sender_id, with a guided fix flow that re-prompts for the four values inline.
- **Reauth flow** — when Ajax rejects the session (password rotated, forced logout, newly-enabled 2FA), the integration now raises `ConfigEntryAuthFailed` and HA shows the orange Reconfigure banner instead of silently failing every poll. This is the OAuth-refresh path you hypothesised about.

If your symptom is "coordinator stuck for hours with no errors anywhere", on 1.2.4 it should now surface as either an `hts_chronic_failure` Repair card or as `last poll: 2h ago` in System Health — visible without log archaeology.

### What we need to triage further

For @Hansontech190 and @jjeremia, reopening this productively needs three pieces of info, since the 1.2.2 fixes already covered the original failure modes:

1. **Integration version** you're running (HACS, or Settings → Devices & Services → Aegis for Ajax → ⋮ → System information). If pre-1.2.2: upgrade to `1.2.4-beta.6` first and reproduce.
2. After the next freeze, **before reloading**: a screenshot of the **System Health card** showing the Aegis section.
3. A **debug log that covers a full freeze cycle** — spans both the moment state was last updated *and* the next 10–15 min of silence. The previous logs in this thread were ~1 second of startup, too narrow to see what the coordinator is doing during the stall.

Enable debug without restart via *Developer Tools → Actions → `logger.set_level`* (YAML mode):

```yaml
custom_components.aegis_ajax: debug
custom_components.aegis_ajax.coordinator: debug
custom_components.aegis_ajax.notification: debug
custom_components.aegis_ajax.api.client: debug
custom_components.aegis_ajax.api.hts.protocol: debug
```

If on 1.2.4-beta.6 the System Health card already shows the symptom (e.g. `last poll: 2h ago`, or HTS streams `0/1`), that screenshot alone is enough — happy to reopen and move it to a clean new bug with a working hypothesis.

### iOS FCM credentials (@jjeremia)

Short answer: there's no practical extraction path from the iOS Ajax app. The `.ipa` ships `GoogleService-Info.plist` inside the signed-and-encrypted app bundle; without a jailbroken device you can't read it.

Practical workaround: **use the Android APK regardless of which OS you actually use day-to-day**. The Firebase project the Ajax app talks to is the same on both platforms (same `project_id`, `gcm_defaultSenderId`, backing keys), so the four values from the Android build work for FCM push delivery on Home Assistant whether you personally use iPhone or Android. The README's "How to obtain FCM credentials" walks the Android path. I'll send a README clarification so this isn't buried in a closed-issue comment.
